### DISCUSSION TITLE
Limiting RecursionDepth with OmitOnRecursionBehavior does not seem to work with LazyEntityGraph

### DISCUSSION THREAD

**Discussion by SubjectiveReality** [AutoFixture/AutoFixture#1111] [Q&A]
```
var fixture = new Fixture();

var customization = new LazyEntityGraphCustomization(
                ModelMetadataGenerator.LoadFromCodeFirstContext(str => new Mydb(), false));

fixture.Customize(customization);

fixture.Behaviors.OfType<ThrowingRecursionBehavior>().ToList().ForEach(b => fixture.Behaviors.Remove(b));
fixture.Behaviors.Add(new OmitOnRecursionBehavior(1)); // set RecursionDepth to 1
```
This does not seem to work as expected, and AutoFixture combined with LazyEntityGraph will recurse through n depth instead of the depth specified in OmitOnRecursionBehavior.  

Perhaps there's another way to do this or I'm missing something.

**Comment by zvirja** [comment]
Hi,

Could you please provide me with a [Minimal, Complete, and Verifiable example](https://stackoverflow.com/help/mcve) of code that helps to reproduce the scenario? The existing sample contains unknown classes, which might be crucial for the issue troubleshooting.

Thanks!
